File: app/src/main/java/wangdaye/com/geometricweather/db/controllers/LocationEntityController.java
package wangdaye.com.geometricweather.db.controllers;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import java.util.List;

import wangdaye.com.geometricweather.db.GeometricWeatherDatabase;
import wangdaye.com.geometricweather.db.entities.LocationEntity;

public class LocationEntityController extends AbsEntityController {

    public static void insertLocationEntity(@NonNull GeometricWeatherDatabase database,
                                            @NonNull LocationEntity entity) {
        database.locationDao().insertLocationEntity(entity);
    }

    public static void insertLocationEntityList(@NonNull GeometricWeatherDatabase database,
                                                @NonNull List<LocationEntity> entityList) {
        if (!entityList.isEmpty()) {
            database.locationDao().insertLocationEntityList(entityList);
        }
    }

    public static void deleteLocationEntity(@NonNull GeometricWeatherDatabase database,
                                            @NonNull LocationEntity entity) {
        database.locationDao().deleteLocationEntity(entity);
    }

    public static void deleteLocationEntityList(@NonNull GeometricWeatherDatabase database) {
        database.locationDao().deleteLocationEntityList();
    }

    public static void updateLocationEntity(@NonNull GeometricWeatherDatabase database,
                                            @NonNull LocationEntity entity) {
        database.locationDao().updateLocationEntity(entity);
    }

    @Nullable
    public static LocationEntity selectLocationEntity(@NonNull GeometricWeatherDatabase database,
                                                      @NonNull String formattedId) {
        return database.locationDao().selectLocationEntity(formattedId);
    }

    @NonNull
    public static List<LocationEntity> selectLocationEntityList(
            @NonNull GeometricWeatherDatabase database) {
        return getNonNullList(database.locationDao().selectLocationEntityList());
    }

    public static int countLocationEntity(@NonNull GeometricWeatherDatabase database) {
        return database.locationDao().countLocationEntity();
    }
}

File: data/src/main/java/wangdaye/com/geometricweather/db/controllers/LocationEntityController.kt
package wangdaye.com.geometricweather.db.controllers

import wangdaye.com.geometricweather.db.GeometricWeatherDatabase
import wangdaye.com.geometricweather.db.entities.LocationEntity

object LocationEntityController {

    @JvmStatic
    fun insertLocationEntity(database: GeometricWeatherDatabase, entity: LocationEntity) {
        database.locationDao().insertLocationEntity(entity)
    }

    @JvmStatic
    fun insertLocationEntityList(
        database: GeometricWeatherDatabase,
        entityList: List<LocationEntity>
    ) {
        if (entityList.isNotEmpty()) {
            database.locationDao().insertLocationEntityList(entityList)
        }
    }

    @JvmStatic
    fun deleteLocationEntity(database: GeometricWeatherDatabase, entity: LocationEntity) {
        database.locationDao().deleteLocationEntity(entity)
    }

    @JvmStatic
    fun deleteLocationEntityList(database: GeometricWeatherDatabase) {
        database.locationDao().deleteLocationEntityList()
    }

    @JvmStatic
    fun updateLocationEntity(database: GeometricWeatherDatabase, entity: LocationEntity) {
        database.locationDao().updateLocationEntity(entity)
    }

    @JvmStatic
    fun selectLocationEntity(
        database: GeometricWeatherDatabase,
        formattedId: String
    ): LocationEntity? {
        return database.locationDao().selectLocationEntity(formattedId)
    }

    @JvmStatic
    fun selectLocationEntityList(database: GeometricWeatherDatabase): List<LocationEntity> {
        return getNonNullList(database.locationDao().selectLocationEntityList())
    }

    @JvmStatic
    fun countLocationEntity(database: GeometricWeatherDatabase): Int {
        return database.locationDao().countLocationEntity()
    }
}
